metadata-ingestion/src/datahub/ingestion/source/sql/presto.py

```diff
@@ -0,0 +1,136 @@
+        raise exc.NoSuchTableError("schema is required")
+
+    if view_name is None:
+        raise exc.NoSuchTableError("view_name is required")
+
+    query = dedent(
+        f"""
+        SHOW CREATE VIEW "{schema}"."{view_name}"
+    """
+    ).strip()
+    res = connection.execute(sql.text(query))
+    return next(res)[0]
+
+
+@reflection.cache  # type: ignore
+def get_table_comment(self, connection, table_name: str, schema: str = None, **kw):  # type: ignore
+    try:
+        properties_table = self._get_full_table(f"{table_name}$properties", schema)
+        query = f"SELECT * FROM {properties_table}"
+        row = connection.execute(sql.text(query)).fetchone()
+
+        # Generate properties dictionary.
+        properties = {}
+        if row:
+            for col_name, col_value in row.items():
+                properties[col_name] = col_value
+
+        return {"text": properties.get("comment", None), "properties": properties}
+    # Fallback to default trino-sqlalchemy behaviour if `$properties` table doesn't exist
+    except Exception:
+        # It seems like on Presto there is no system.metada.table_comment table which exists on Trino
+        # The best thing we can do is to return empty
+        # For views this will be empty on Presto (or at least this is what I can see)
+        return {}
+
+
+def _get_full_table(  # type: ignore
+    self, table_name: str, schema: str = None, quote: bool = True
+) -> str:
+    table_part = (
+        self.identifier_preparer.quote_identifier(table_name) if quote else table_name
+    )
+    if schema:
+        schema_part = (
+            self.identifier_preparer.quote_identifier(schema) if quote else schema
+        )
+        return f"{schema_part}.{table_part}"
+
+    return table_part
+
+
+PrestoDialect.get_table_names = get_table_names
+PrestoDialect.get_view_names = get_view_names
+PrestoDialect.get_view_definition = get_view_definition
+PrestoDialect.get_table_comment = get_table_comment
+PrestoDialect.get_columns = _get_columns
+PrestoDialect._get_full_table = _get_full_table
+
+
+class PrestoConfig(TrinoConfig):
+    # defaults
+    scheme = Field(default="presto", description="", hidden_from_schema=True)
+
+
+@platform_name("Presto", doc_order=1)
+@config_class(PrestoConfig)
+@support_status(SupportStatus.CERTIFIED)
+@capability(SourceCapability.DOMAINS, "Supported via the `domain` config field")
+@capability(SourceCapability.DATA_PROFILING, "Optionally enabled via configuration")
+class PrestoSource(TrinoSource):
+    """
+
+    This plugin extracts the following:
+
+    - Metadata for databases, schemas, and tables
+    - Column types and schema associated with each table
+    - Table, row, and column statistics via optional SQL profiling
+
+    """
+
+    config: PrestoConfig
+
+    def __init__(self, config, ctx):
+        super().__init__(config, ctx)
+        # We have to override platform as this source inherits from the Trino source
+        self.platform = "presto"
```

Comment:
can we make the trino source take an optional platform param in its init method?